oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/Version.java

```diff
@@ -35,14 +61,28 @@ public Version(int major, int minor, int patch, int update, int build)
      */
     public static Version parseJavaRuntimeVersion(final String javaRuntimeVersion)
     {
-        Matcher matcher = getMatches(JAVA_RUNTIME_VERSION, javaRuntimeVersion);
-        int major = Integer.parseInt(matcher.group(1));
-        int minor = Integer.parseInt(matcher.group(2));
-        int patch = integerOrZero(matcher.group(3));
-        int update = integerOrZero(matcher.group(4));
-        int build = integerOrZero(matcher.group(5));
-
-        return new Version(major, minor, patch, update, build);
+        try {
+            Matcher matcher = getMatches(JAVA_9_PLUS_VERSION, javaRuntimeVersion);

```

Comment:
`getMatches()` doesn't do much, the only thing it does is to compile the pattern and throw exception.  If we use that function, then unfortunately we must control flow with exceptions.  Would you mind change this to something like:
```
Matcher matcher = JAVA_9_PLUS_VERSION.matcher(javaRuntimeVersion);
if (matcher.matches()) {
  // proceed 
} else {
    matcher = JAVA_RUNTIME_VERSION.matcher(javaRuntimeVersion);
    if (matcher.matches()) {
        ....
    } else {
        //throw IllegalArgumentException here.
    }
}
```
Also, the name of `JAVA_RUNTIME_VERSION` seems no longer accurate, can we update it to `JAVA_LEGACY_RUNTIME_VERSION`?